### PR TITLE
add examples for L7-XLB Header & Parameter Based Routing

### PR DESCRIPTION
**Pull request by edwardmedia**
Fixes https://github.com/terraform-providers/terraform-provider-google/issues/6340

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none
```

### CODE REVIEW

File: templates/terraform/examples/url_map_header_based_routing.tf.erb [hunk 1]
```diff
@@ -0,0 +1,78 @@
+resource "google_compute_url_map" "<%= ctx[:primary_resource_id] %>" {
+name        = "<%= ctx[:vars]['url_map_name'] %>"
```
Comment (danawillow) [inline templates/terraform/examples/url_map_header_based_routing.tf.erb:2]:
```suggestion
  name        = "<%= ctx[:vars]['url_map_name'] %>"
```

File: templates/terraform/examples/url_map_header_based_routing.tf.erb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  health_checks = [google_compute_http_health_check.default.id]
+}
+
+resource "google_compute_http_health_check" "default" {
+  name               = "<%= ctx[:vars]['health_check_name'] %>"
+  request_path       = "/"
+  check_interval_sec = 1
+  timeout_sec        = 1
+}
+
```
Comment (danawillow) [inline templates/terraform/examples/url_map_header_based_routing.tf.erb:75]: can you remove these extra newlines? one is sufficient.

File: templates/terraform/examples/url_map_parameter_based_routing.tf.erb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  health_checks = [google_compute_http_health_check.default.id]
+}
+
+resource "google_compute_http_health_check" "default" {
+  name               = "<%= ctx[:vars]['health_check_name'] %>"
+  request_path       = "/"
+  check_interval_sec = 1
+  timeout_sec        = 1
+}
+
```
Comment (danawillow) [inline templates/terraform/examples/url_map_parameter_based_routing.tf.erb:75]: same here

File: templates/terraform/examples/url_map_header_based_routing.tf.erb (hunk 1 above)
Comment (edwardmedia) [reply templates/terraform/examples/url_map_header_based_routing.tf.erb]: Fixed in another commit

File: templates/terraform/examples/url_map_parameter_based_routing.tf.erb (hunk 3 above)
Comment (edwardmedia) [reply templates/terraform/examples/url_map_parameter_based_routing.tf.erb:75]: Fixed in another commit

File: templates/terraform/examples/url_map_header_based_routing.tf.erb (hunk 2 above)
Comment (edwardmedia) [reply templates/terraform/examples/url_map_header_based_routing.tf.erb:75]: Fixed in another commit

**Pull request merged**